vms/rpcchainvm/vm.go

```diff
@@ -43,20 +43,11 @@ func Serve(ctx context.Context, vm block.ChainVM, opts ...grpcutils.ServerOption
 			fmt.Println("vm server: graceful termination success")
 		}()
 
-		for {
```

Comment:
We return on receipt of any any of the handled signals or context cancellation so no need for loop